custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -109,5 +109,20 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
                 (selectedLocations[level - 1] && selectedLocations[level - 1] !== 'all' && selectedLocations[level - 1].location_id !== 'all')
             );
         },
+        getLocationsForLevel: function(level, selectedLocations, locationsCache) {
```

Comment:
This is another commit that could affect web dashboard so should be reviewed with care. though again, i tried to be quite meticulous in refactoring and testing.